src/k2/persistence/plog_service/PlogServer.cpp

```diff
@@ -0,0 +1,169 @@
+        return handleRead(std::move(request));
+    });
+
+    RPC().registerRPCObserver<dto::PlogSealRequest, dto::PlogSealResponse>(dto::Verbs::K23SI_PERSISTENT_SEAL, [this](dto::PlogSealRequest&& request) {
+        return handleSeal(std::move(request));
+    });
+    _plogMap.clear();
+
+    return seastar::make_ready_future<>();
+}
+
+seastar::future<std::tuple<Status, dto::PlogCreateResponse>>
+PlogServer::handleCreate(dto::PlogCreateRequest&& request){
+    K2DEBUG("Received create request for " << request.plogId);
+    auto iter = _plogMap.find(request.plogId);
+    if (iter != _plogMap.end()) {
+        return RPCResponse(Statuses::S400_Bad_Request("plog already exists"), dto::PlogCreateResponse());
+    }
+    _plogMap.insert(std::pair<String,PlogPage >(std::move(request.plogId), PlogPage()));
+    return RPCResponse(Statuses::S201_Created("plog created"), dto::PlogCreateResponse());
+};
+
+seastar::future<std::tuple<Status, dto::PlogAppendResponse>>
+PlogServer::handleAppend(dto::PlogAppendRequest&& request){
+    K2DEBUG("Received append request for " << request.plogId << " with size" << request.payload.getSize() << " and offset " << request.offset);
+    auto iter = _plogMap.find(request.plogId);
+    if (iter == _plogMap.end()) {
+        return RPCResponse(Statuses::S400_Bad_Request("plog does not exist"), dto::PlogAppendResponse());
+    }
+    if (iter->second.sealed){
+         return RPCResponse(Statuses::S400_Bad_Request("plog is sealed"), dto::PlogAppendResponse());
+    }
+    if (iter->second.offset != request.offset){
+        return RPCResponse(Statuses::S400_Bad_Request("offset inconsistent"), dto::PlogAppendResponse());
+    }
+    if (iter->second.offset + request.payload.getSize() > PLOG_MAX_SIZE){
+         return RPCResponse(Statuses::S400_Bad_Request("exceeds pLog limit"), dto::PlogAppendResponse());
+    }
+
+    dto::PlogAppendResponse response;
+    response.offset = iter->second.offset + request.payload.getSize();
+    response.bytes_appended = request.payload.getSize();
+
+    iter->second.offset += request.payload.getSize();
+    iter->second.payload.copyFromPayload(request.payload, request.payload.getSize());
+    request.payload.clear();
+    
+    return RPCResponse(Statuses::S200_OK("append cuccess"), std::move(response));
+};
+
+
+seastar::future<std::tuple<Status, dto::PlogReadResponse>>
+PlogServer::handleRead(dto::PlogReadRequest&& request){
+    K2DEBUG("Received read request for " << request.plogId << " with offset " << request.offset);
+    auto iter = _plogMap.find(request.plogId);
+    if (iter == _plogMap.end()) {
+        return RPCResponse(Statuses::S400_Bad_Request("plog does not exist"), dto::PlogReadResponse());
+    }
+    if (iter->second.offset < request.offset + request.size){
+         return RPCResponse(Statuses::S400_Bad_Request("exceed the maximun length"), dto::PlogReadResponse());
+    }
+    
+    Payload payload([size=request.size] { return Binary(size); });

```

Comment:
reads shouldn't require a copy.I think we can enhance Payload to provide a method to share data into another payload:
```c++
Payload Payload::share(size_t nbytes) 
```

Reply:
I added the Payload Payload::share(size_t nbytes) in Payload.h and fixed this problem